File: hw/codena/u3/hw24CompositePatterCode/Python/composite_pattern/model/employee.py
from abc import ABC


class Employee(ABC):

    def __init__(self):
        self.name = "Not assigned yet"
        self.title = "Not assigned yet"

    def stateName(self):
        print(f"{self.title} {self.name}")

    def add(self, employee):
        raise NotImplementedError("Leaf objects cannot add employees.")

    def remove(self, employee):
        raise NotImplementedError("Leaf objects cannot remove employees.")

    def getChild(self, index):
        raise NotImplementedError("Leaf objects do not have children.")

File: hw/codena/u3/hw24CompositePatterCode/Javascript/CompositePattern/model/Employee.js
export default class Employee {

    constructor() {
        this.name = "Not assigned yet";
        this.title = "Not assigned yet";
    }

    stateName() {
        console.log(`${this.title} ${this.name}`);
    }

    add(employee) {
        throw new Error("Leaf objects cannot add employees.");
    }

    remove(employee) {
        throw new Error("Leaf objects cannot remove employees.");
    }

    getChild(index) {
        throw new Error("Leaf objects do not have children.");
    }

}